app/src/main/java/com/github/k409/fitflow/ui/screen/profile/ProfileScreen.kt

```diff
@@ -0,0 +1,274 @@
+                            },
+                        imageVector = Icons.Outlined.Info,
+                        contentDescription = null,
+                    )
+                }
+            }
+
+            item(span = {
+                GridItemSpan(maxLineSpan)
+            }) {
+                WidgetCard(title = stringResource(R.string.level_progress)) {
+                    Text(
+                        text = if (maxXp == Int.MAX_VALUE) {
+                            "$minXp+"
+                        } else if (progress >= 1) {
+                            "$maxXp / $maxXp"
+                        } else {
+                            "$userXp / $maxXp"
+                        },
+                        style = MaterialTheme.typography.labelLarge,
+                        fontWeight = FontWeight.Normal,
+                        modifier = Modifier.padding(end = 2.dp),
+                    )
+
+                    val progressColor =
+                        if (progress >= 1) {
+                            MaterialTheme.colorScheme.primary
+                        } else MaterialTheme.colorScheme.primary.copy(
+                            alpha = 0.5f,
+                        )
+
+                    LinearProgressIndicator(
+                        progress = { progress },
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(end = 6.dp, top = 2.dp)
+                            .height(12.dp)
+                            .clip(RoundedCornerShape(8.dp)),
+                        color = progressColor,
+                    )
+                }
+            }
+
+            item {
+                WidgetCard(
+                    title = stringResource(id = R.string.points),
+                    value = currentUser.points.toString(),
+                )
+            }
+
+            item {
+                WidgetCard(
+                    title = stringResource(R.string.height_cm),
+                    value = currentUser.height.toString(),
+                )
+            }
+
+            item {
+                WidgetCard(
+                    title = stringResource(R.string.weight_kg),
+                    value = currentUser.weight.toString(),
+                )
+            }
+
+            item {
+                WidgetCard(title = stringResource(R.string.gender), value = currentUser.gender)
+            }
+
+            item {
+                WidgetCard(
+                    title = stringResource(R.string.birth_date),
+                    value = currentUser.dateOfBirth,
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun WidgetCard(
+    modifier: Modifier = Modifier,
+    title: String,
+    value: String,
+    actions: @Composable RowScope.() -> Unit = {},
+) {
+    WidgetCard(
+        modifier = modifier,
+        title = title,
+        actions = actions,
+    ) {
```

Comment:
Maybe rename the inner WidgetCard function for better readability.

Reply:
Inner WidgetCard and outside one have the same functionality, just different method overload. Unless you have a suggestion for better name.

Reply:
I though about something like WidgetCardWrapper, but since it works more as an overload, then its all good.